frontend/src/components/Dish/DishIndex.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState, useContext } from "react";
+import DishItem from "./DishItem";
+import userContext from "../../contexts/context";
+
+const _ = require("lodash");
+
+const apiUtil = require("../../util/apiUtil");
+
+
+function DishIndex(props) {
+  const [dishesState, setDishes] = useState(() => [])
+  const isAdmin = useContext(userContext).admin;
+
+  function getDishesSucces(dishes) {
+      
+      let array = [];
+      array.push(<DishItem isAdmin={isAdmin} formType={"new"} setFormInfo={props.setFormInfo} setFormCallback={props.setFormCallback} />);
+    _.forEach(dishes, dish => {
```

Comment:
As I told you before, you almost never need loops in JS.
Always use [lodash](https://lodash.com/docs/4.17.15) - it's a 3rd party library that has all the functions you'll need.
Also - never use State if you don't *have* to. in this case, you can just render the array directly, no need to save it to a state. This component is probably redundant, but if you keep it, then it should get the array of dishes in the props - and the state in which we keep the data that returns from an api call, should be one layer above it - in the Console component. The higher Console component should do all the API calls and save the data in a formatted way (like the components expect to get it in the props), and then each component will take what it needs from the context, or you can pass them the data in the props.
Regarding lodash - In this example, you could just do:
```
return _.map(props.dishes, (dish) => <DishItem dish={dish} isAdmin={isAdmin} ... />);
```
But I really think that since this whole component is basically a `map` inside a `div` that will have a styling - you don't need it. It can be placed inline in a `BranchPage` or `BusinessPage`